eslint.config.js

```diff
@@ -78,6 +78,23 @@ function createInternalFilesPatterns(pattern = null) {
 
 module.exports = [
     ...compat.extends("eslint"),
+    {
+        ignores: [
+            "build/**",
+            "coverage/**",
+            "docs/**",
+            "!docs/.eleventy.js",
+            "jsdoc/**",
+            "templates/**",
+            "tests/bench/**",
+            "tests/fixtures/**",
+            "tests/performance/**",
+            "tmp/**",
+            "tools/internal-rules/node_modules/**",
+            "**/test.js",
+            "!**/.eslintrc.js"
```

Comment:
Just double-checking: this still doesn't unignore `.eslintrc.js` correct?

Reply:
Correct. When doing a glob search with a pattern that doesn't have an explicit dot that matches leading dots in file names, `.eslintrc.js` will still be ignored because globby doesn't support negative patterns in `ignore` (in fact, I believe it removes leading `!` so this actually ignores `**/.eslintrc.js`), and in general globby's dotfiles matching behavior seems to depend only on the content of search patterns.

On the other hand, when directly using our config array to evaluate whether or not `.eslintrc.js` is ignored (e.g., when passing a file path), it's _not_ ignored regardless of this line. This behavior needs further analysis (https://github.com/eslint/eslint/issues/16265).

Overall, this line currently does not affect anything, but it may when we fix https://github.com/eslint/eslint/issues/16265 so I've left it in for now.